File: framework/test2/common/intersect/Intersect.scala
import soda.scala.unittest._

import scala.collection.mutable
import scala.reflect.runtime.universe.typeOf

class Solution {}
object Solution {
  def intersection(nums1: Array[Int], nums2: Array[Int]): Array[Int] = {
    if (nums1.size > nums2.size) {
      return intersection(nums2, nums1)
    }

    val mset = mutable.Set[Int]()
    val res = mutable.Set[Int]()
    for (n <- nums1) {
      mset += n
    }
    for (b <- nums2) {
      if (mset.contains(b)) {
        res += b
      }
    }
    res.toArray
  }
}

class Intersect {
  def get(): TestWork = {
    val work = new TestWork(typeOf[Solution], "intersection")
    // val work = TestWork.forStruct(...)
    work.setValidator(Validators.forArray[Int](false))
    work.compareSerial = true
    work
  }
}

object Intersect extends App {
  new Intersect().get().run()
}

File: framework/test2/common/intersect/Intersect.kt
import soda.kotlin.unittest.*

class Solution {
    fun intersection(nums1: IntArray, nums2: IntArray): IntArray {
        if (nums1.size > nums2.size) {
            return intersection(nums2, nums1)
        }

        val mset = mutableSetOf<Int>()
        val res = mutableSetOf<Int>()
        for (n in nums1) {
            mset += n
        }
        for (b in nums2) {
            if (b in mset) {
                res += b
            }
        }
        return res.toIntArray()
    }
}

class Intersect : (String) -> String {
    override fun invoke(text: String): String {
        // * setup with your method
        val work = GenericTestWork.create(Solution()::intersection)

        // * by method that has not return value
        // val work = GenericTestWork.createVoid(Solution.METHOD_WITHOUT_RETURN)
        // * by class of data struct
        // val work = GenericTestWork.forStruct(STRUCT::class)

        // * setup validator
        work.validator = Validators.forIntArray(false)
        // work.compareSerial = true
        return work.run(text)
    }

}

fun main() {
    print(Intersect()(Utils.fromStdin()))
}
